src/cocotb/regression.py

```diff
@@ -134,28 +172,75 @@ async def func(*args, **kwargs):
         self.stage = stage
         self.name = self.func.__qualname__ if name is None else name
         self.module = self.func.__module__ if module is None else module
+        self.doc = self.func.__doc__ if doc is None else doc
+        if self.doc is not None:
+            # cleanup docstring using `trim` function from PEP257
+            self.doc = inspect.cleandoc(self.doc)
         self.fullname = f"{self.module}.{self.name}"
-        self.included: Optional[bool] = None
+
+
+def _format_doc(docstring: Union[str, None]) -> str:
+    if docstring is None:
+        return ""
+    else:
+        brief = docstring.split("\n")[0]
+        return f"\n    {brief}"
+
+
+class RegressionMode(DocEnum):
+    """The mode of the :class:`RegressionManager`."""
+
+    REGRESSION = (
+        auto(),
+        """Tests are run if included. Skipped tests are skipped, expected failures and errors are respected.""",
+    )
+
+    TESTCASE = (
+        auto(),
+        """Like :attr:`REGRESSION`, but skipped tests are *not* skipped if included.""",
+    )
 
 
 class RegressionManager:
-    """Encapsulates all regression capability into a single place"""
+    """Object which manages tests.
+
+    This object uses the builder pattern to build up a regression.
+    Tests are added using :meth:`register_test` or :meth:`discover_tests`.
+    Filters for tests can be added using :meth:`add_filters`.
```

Comment:
```suggestion
    Inclusion filters for tests can be added using :meth:`add_filters`.
```
maybe? IMO we almost can't be explicit enough with this word...